packages/xo-web/src/xo-app/backup-ng/new/index.js

```diff
@@ -553,7 +554,22 @@ export default [
       }),
       srPredicate: ({ srs }) => sr => isSrWritable(sr) && !includes(srs, sr.id),
       remotePredicate: ({ remotes }) => ({ id }) => !includes(remotes, id),
-      propSettings: (_, { job }) => Map(get(() => job.settings)),
+      propSettings: (_, { job }) =>
+        Map(get(() => job.settings)).map(
+          setting =>
+            defined(
+              setting.copyRetention,
+              setting.exportRetention,
+              setting.snapshotRetention
+            )
+              ? {
+                  copyRetention: setting.copyRetention || DEFAULT_RETENTION,
+                  exportRetention: setting.exportRetention || DEFAULT_RETENTION,
+                  snapshotRetention:
+                    setting.snapshotRetention || DEFAULT_RETENTION,
+                }
+              : setting
+        ),
```

Comment:
Can you explain this?

Reply:
On editing the backup, i set all retention to 1 when it not exists and on saving the form i normalise it.

Reply:
Please [see](https://github.com/vatesfr/xen-orchestra/pull/3290/files#diff-9c044ab8a42ed6576ea927a64c1ec3ebR51)

Reply:
So `setting` only has `copyRetention`, `exportRetention` and `snapshotRetention` at this point? Or are you overriding something?
Why don't you simply do:
```js
setting => ({
  copyRetention: setting.copyRetention || DEFAULT_RETENTION,
  exportRetention: setting.exportRetention || DEFAULT_RETENTION,
  snapshotRetention: setting.snapshotRetention || DEFAULT_RETENTION,
})
```

Reply:
So in case `defined(...)` is truthy, you'll be overriding those settings, right?

Reply:
Yes, true.

Reply:
Is that what you want?